Formula/qt.rb

```diff
@@ -27,6 +28,10 @@ class Qt < Formula
   uses_from_macos "flex"
   uses_from_macos "sqlite"
 
+  # Fix build on Linux when the build system has AVX2
+  # Patch submitted at https://bugreports.qt.io/browse/QTBUG-84851
```

Comment:
This should be submitted to https://codereview.qt-project.org. In my experience a patch on their Jira has almost no chance of being merged.

Reply:
https://codereview.qt-project.org/c/qt/qt3d/+/303990

That is my first submission of a patch to Guerrit. I hope it is fine.

Reply:
Great work, thanks ❤️. Could you update the comment to add that URL and change the patch to be a `url` with `https://codereview.qt-project.org/gitweb?p=qt/qt3d.git;a=patch;h=b456a7d47a36dc3429a5e7bac7665b12d257efea` and the relevant `sha256`? Thanks again!

Reply:
I cannot (I tried), because that patch is against `qt3d` whereas I need a patch against the qt5 sources as a whole.

Reply:
You can change the patch directory with e.g. `directory "tools/build"` in the `patch do` block.

Reply:
Something like
```ruby
patch do
  url "https://codereview.qt-project.org/gitweb?p=qt/qt3d.git;a=patch;h=b456a7d47a36dc3429a5e7bac7665b12d257efea"
  sha256 "..."
  directory "qt3d"
end
```

Reply:
Thanks! Done in https://github.com/Homebrew/homebrew-core/pull/55964/commits/e25f45416fd7a5848af11d61b5be5aa7be8a6367, and I have tested that it works.